src/evaluators/ThrowStatement.js

```diff
@@ -27,8 +27,12 @@ export default function(
   let exprValue = Environment.GetValue(realm, exprRef);
   if (realm.isInPureScope() && !realm.isInPureTryStatement) {
     invariant(realm.generator !== undefined);
+    // TODO: we should porbably materialize exprValue at this point
     realm.generator.emitThrow(exprValue);
-    throw new PureScopeThrowCompletion(realm, ast.loc);
+    // We want a completion to bubble up the execution path, but
+    // we don't want the completion to serialize to a value as
+    // we do generate a generator entry above instead.
+    throw new SimpleNormalCompletion(realm.intrinsics.empty, ast.loc);
```

Comment:
Throwing a normal completion smells wrong. I would be awesome if you make this work by using a ThrowCompletion.

Reply:
Will try using a flag instead.